docs/manifests_and_customizing_via_api.md

```diff
@@ -0,0 +1,335 @@
+    zone: us-east-2c
+  topology:
+    bastion:
+      bastionPublicName: bastion.k8s.example.com
+    dns:
+      type: Public
+    masters: private
+    nodes: private
+
+---
+
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:48Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: bastions
+spec:
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: t2.micro
+  maxSize: 1
+  minSize: 1
+  role: Bastion
+  subnets:
+  - utility-us-east-2d
+  - utility-us-east-2b
+  - utility-us-east-2c
+
+
+---
+
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:47Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: master-us-east-2d
+spec:
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: m4.large
+  maxSize: 1
+  minSize: 1
+  role: Master
+  subnets:
+  - us-east-2d
+
+
+---
+
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:47Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: master-us-east-2b
+spec:
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: m4.large
+  maxSize: 1
+  minSize: 1
+  role: Master
+  subnets:
+  - us-east-2b
+
+
+---
+
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:48Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: master-us-east-2c
+spec:
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: m4.large
+  maxSize: 1
+  minSize: 1
+  role: Master
+  subnets:
+  - us-east-2c
+
+
+---
+
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:48Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: nodes
+spec:
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: m4.xlarge
+  maxSize: 3
+  minSize: 3
+  role: Node
+  subnets:
+  - us-east-2d
+  - us-east-2b
+  - us-east-2c
+```
+
+Next, delete the cluster from the state store.
+
+```
+kops delete -f $NAME.yaml
+# validate that you want to remove the cluster
+kops delete -f $NAME.yaml --yes
+```
+
+## YAML examples
+
+FIXME: rename this section.
+
+With the above YAML file, a user can add configurations that are not available via the command line.  For instance, you can add a `MaxPrice` value to a new instance group and use spot instances.  Also add node and cloud labels for the new instance group.
+
+```
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:48Z
+  labels:
+    kops.k8s.io/cluster: k8s.example.com
+  name: my-crazy-big-nodes
+spec:
+ nodeLabels:
+    spot: "true"
+  cloudLabels:
+    team: example
+    project: ion
+  image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-05-02
+  machineType: m4.10xlarge
+  maxSize: 42
+  minSize: 42
+  maxPrice: "0.35"
+  role: Node
+  subnets:
+  - us-east-2d
```

Comment:
If your node ASG spans AZs it might autoscale a node but k8s won't be able to schedule the pod it did it for because it's got an EBS volume in a different AZ.